### PR TITLE
bpo-44953: Add vectorcall for itemgetter and attrgetter

### PR DESCRIPTION
**Pull request by sweeneyde**
https://bugs.python.org/issue44953

### CODE REVIEW

**Comment by sweeneyde** [comment]
This was inspired by GH-27782, which does the same, and more, for methodcaller.

**Comment by sweeneyde** [comment]
Benchmarks for this change:
```
itemgetter: Mean +- std dev: [operator_main] 45.3 ns +- 1.3 ns -> [operator_vec] 29.5 ns +- 0.7 ns: 1.54x faster
attrgetter: Mean +- std dev: [operator_main] 61.6 ns +- 1.7 ns -> [operator_vec] 43.8 ns +- 0.9 ns: 1.41x faster
```

**Review by rhettinger — APPROVED** [review]

File: Modules/_operator.c
Comment (markshannon) [inline Modules/_operator.c:1160]: Why are you adding this attribute?

File: Modules/_operator.c
Comment (markshannon) [inline Modules/_operator.c:1524]: Ditto

File: Modules/_operator.c
Comment (markshannon) [reply Modules/_operator.c:1160]: Apparently this is the way to add vectorcall capability :disappointed: 
So, leave this and I'll open an issue to add the vectorcall capability a sensible way.

**Comment by sweeneyde** [comment]
The `__vectorcalloffset__` behavior was added in GH-20026. I don't think I understand that design decision rather than adding a new  `PyType_Slot.slot` value called something like `Py_tp_vectorcall`, but it appears to be what we have. The corresponding behavior of `__dictoffset__` and `__weaklistoffset__` goes back to 2019 in GH-16076. Are there any plans to change these, or is there anything holding this up?

**Review by methane — APPROVED** [review]

**Comment by sweeneyde** [comment]
Any plans for this? It may also be relevant to bpo-46148 about optimizing Pathlib.

**Comment by sweeneyde** [comment]
Does anyone with a commit bit want to merge this? This is the kind of callable that's often called in loops.

**Comment by sweeneyde** [comment]
I re-ran some benchmarks, and they still look good. I plan to merge this in about 24 hours if there are no more objections.

```python
from operator import itemgetter, attrgetter
from itertools import repeat
from collections import namedtuple, deque
from pyperf import Runner

class DefaultClass:
    def __init__(self, a, b):
        self.a = a
        self.b = b

class SlotsClass:
    __slots__ = "a", "b"
    def __init__(self, a, b):
        self.a = a
        self.b = b

NamedTuple = namedtuple("NT", ["a", "b"])
MAP_LOOPS = 10_000

attr_classes = {
    'DefaultClass': DefaultClass,
    'SlotsClass': SlotsClass,
    'NamedTuple': NamedTuple,
}

item_classes = {
    'tuple': tuple,
    'list': list,
    'dict': dict.fromkeys,
}

namespace = {
    'IG': itemgetter(1),
    'AG': attrgetter('a'),
    'repeat': repeat,
    'deque': deque,
} | attr_classes | item_classes

runner = Runner()

for classname in attr_classes:
    runner.timeit(
        name=f"{classname}-1",
        setup=f"obj = {classname}(11, 22)",
        stmt="AG(obj)",
        globals=namespace
    )
    runner.timeit(
        name=f"{classname}-map",
        setup=f"obj = {classname}(11, 22)",
        stmt=f"deque(map(AG, repeat(obj, {MAP_LOOPS})), maxlen=0)",
        globals=namespace,
        inner_loops=MAP_LOOPS,
    )

for classname in item_classes:
    runner.timeit(
        name=f"{classname}-1",
        setup=f"obj = {classname}((1, 2, 3, 4, 5))",
        stmt="IG(obj)",
        globals=namespace,
    )
    runner.timeit(
        name=f"{classname}-map",
        setup=f"obj = {classname}((1, 2, 3, 4, 5))",
        stmt=f"deque(map(IG, repeat(obj, {MAP_LOOPS})), maxlen=0)",
        globals=namespace,
        inner_loops=MAP_LOOPS,
    )
```

Results:
| Benchmark        | main_bench | PR-27828              |
|------------------|:----------:|:---------------------:|
| tuple-map        | 27.6 ns    | 13.2 ns: 2.09x faster |
| dict-map         | 39.4 ns    | 24.2 ns: 1.62x faster |
| NamedTuple-map   | 40.0 ns    | 24.8 ns: 1.61x faster |
| list-map         | 38.2 ns    | 23.8 ns: 1.61x faster |
| SlotsClass-map   | 42.8 ns    | 28.1 ns: 1.53x faster |
| DefaultClass-map | 45.5 ns    | 31.0 ns: 1.47x faster |
| tuple-1          | 56.7 ns    | 41.4 ns: 1.37x faster |
| NamedTuple-1     | 70.1 ns    | 52.7 ns: 1.33x faster |
| dict-1           | 69.4 ns    | 52.2 ns: 1.33x faster |
| SlotsClass-1     | 75.1 ns    | 56.6 ns: 1.33x faster |
| list-1           | 68.7 ns    | 52.1 ns: 1.32x faster |
| DefaultClass-1   | 76.9 ns    | 59.4 ns: 1.29x faster |
| Geometric mean   | (ref)      | 1.48x faster          |

**Pull request merged**
